test/promisified/consumer/consumeMessages.spec.js

```diff
@@ -410,8 +417,24 @@ describe.each(cases)('Consumer - partitionsConsumedConcurrently = %s -', (partit
             topic: topicName,
             partitions: partitions,
         });
-        await consumer.connect();
+
+        /* Reconfigure producer and consumer in such a way that batches are likely
+         * to be small and we get multiple partitions in the cache at once.
+         * This is to reduce flakiness. */
+        producer = createProducer({}, {
+            'linger.ms': 1,
```

Comment:
It's ensured better  with:
```suggestion
           'batch.num.messages': 1,
```